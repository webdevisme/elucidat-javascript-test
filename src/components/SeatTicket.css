.Ticket {
    position: relative;
    width: 50%;
    margin: 0 25%;
    border: solid 3px black;
    padding: 30px 10px;
    text-align: center;
}
.Ticket img {
    width: 100%;
    height: 60px;
    display: block;
    margin-bottom: 40px;
}
.Ticket .close {
    position: absolute;
    top: 0;
    right: 0;
}
.Ticket table {
    display: inline-table;
}
.Ticket td {
    text-align: left;
}

.Ticket td:nth-of-type(2) {
    text-align: right;
}
Button {
    background-color: transparent;
    border: transparent;
}
table Button {
    margin-top: 40px;
    width: 100%;
    border: solid 2px #333;
}