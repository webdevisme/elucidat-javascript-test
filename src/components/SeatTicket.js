import React from "react";
import './SeatTicket.css';

function SeatTicket(props) {

    const handleClose = (event) => {
        event.preventDefault();

        if (props.close) {
            props.close();
        }
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        props.book(props.seat);
    }
  return (
    <div className="Ticket">
        <button className="close" onClick={handleClose}>Close</button>
        <table className="details">
            <tbody>
                <tr><td>Seat</td><td>{props.seat.seatNumber}</td></tr>
                <tr><td>Available</td><td>{props.seat.available ? 'Yes' : 'No'}</td></tr>
                <tr><td>Disability Access</td><td>{props.seat.disabilityAccessible ? 'Yes' : 'No'}</td></tr>
                <tr><td>Price</td><td>{props.seat.price}</td></tr>
                <tr><td colSpan="2">        <button  onClick={handleSubmit}>Book Seat</button></td></tr>
            </tbody>
        </table>
    </div>
  );
}

export default SeatTicket;
