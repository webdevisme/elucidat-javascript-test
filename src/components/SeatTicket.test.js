import React from 'react';
import renderer from 'react-test-renderer';
import SeatTicket from './SeatTicket';
import seatsData from '../../public/seats.json';

it('renders ticket when seat data passed to prop', () => {
  const component = renderer.create(<SeatTicket seat={seatsData[0]}></SeatTicket>);
  
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
