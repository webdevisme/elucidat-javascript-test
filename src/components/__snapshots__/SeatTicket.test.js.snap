// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ticket when seat data passed to prop 1`] = `
<div
  className="Ticket"
>
  <button
    className="close"
    onClick={[Function]}
  >
    X
  </button>
  <img
    alt="Seat Icon"
    src="Seat.svg"
  />
  <table
    className="details"
  >
    <tbody>
      <tr>
        <td>
          Seat
        </td>
        <td>
          1A
        </td>
      </tr>
      <tr>
        <td>
          Available
        </td>
        <td>
          Yes
        </td>
      </tr>
      <tr>
        <td>
          Disability Access
        </td>
        <td>
          Yes
        </td>
      </tr>
      <tr>
        <td>
          Price
        </td>
        <td>
          Â£19.99
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
                  
          <button
            onClick={[Function]}
          >
            Book Seat
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
